examples/train-text-from-scratch/train-text-from-scratch.cpp

```diff
@@ -302,9 +302,9 @@ void init_model(struct my_llama_model * model) {
         // ggml_set_name(layer.w2, (layers_i + ".feed_forward.w2.weight").c_str());
         // ggml_set_name(layer.w3, (layers_i + ".feed_forward.w3.weight").c_str());
 
-        strncpy(layer.w1->name, (layers_i + ".feed_forward.w1.weight").c_str(), sizeof(layer.w1->name));
-        strncpy(layer.w2->name, (layers_i + ".feed_forward.w2.weight").c_str(), sizeof(layer.w2->name));
-        strncpy(layer.w3->name, (layers_i + ".feed_forward.w3.weight").c_str(), sizeof(layer.w3->name));
+        ggml_format_name(layer.w1, (layers_i + ".feed_forward.w1.weight").c_str());
```

Comment:
The idea would be to do something like this:
```c++
gml_format_name(layer.w1, "%s.feed_forward.w1.weight", layers_i.c_str());
```
However, as @ggerganov said, this won't work due to the maximum size of 32 characters of the names.

I want to strongly reiterate that it is not ok to work around this by using a non-nul terminated string. We need to increase the max name length or otherwise fix it in a way that respects the buffer sizes.

Reply:
I was going to suggest just trimming it down to feed_fwd, but some debug printouts tell me these names have to exactingly match names that I assume are in the model files

Seems like name length is determined by GGML_MAX_NAME, don't know what the consequences of fiddling that are. Bumping it up to any higher multiple of 2 should put it out of range of any reasonable potential name length. Changing it doesn't change anything, looking at output and places related to the name field it seems like there's quite a few places that need to be changed to change the effective name length, stripped this stuff out for now and I'll figure it out in the morning and it'll be a second commit.

Reply:
Increasing `GGML_MAX_NAME` should work. I would suggest increasing it by 16 so that the padding doesn't have to be changed.

Reply:
Alright, I figured out why I thought more needed to be changed beyond `GGML_MAX_NAME`, turns out this change is breaking with models trained using the smaller name limit. Well, I don't know how much breakage it is exactly, the training script still seems to run when I drop the safety assert and feed in an old model file but if a similar safety check is being run anywhere else then bumping `GGML_MAX_NAME` is going to make model files that had previously truncated names stop. Well, I've added the commit to the pr. Currently this only cleans up a compiler warning about a sketchy string copy, if there's potential breakage maybe the name length bump can be left until a day comes when a necessary breaking change happens that the second commit can be folded in alongside.